src/rgw/store/sfs/sfs_gc.cc

```diff
@@ -0,0 +1,173 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t
+// vim: ts=8 sw=2 smarttab ft=cpp
+/*
+ * Ceph - scalable distributed file system
+ * SFS SAL implementation
+ *
+ * Copyright (C) 2022 SUSE LLC
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ */
+#include "sfs_gc.h"
+
+#include "store/sfs/sqlite/sqlite_objects.h"
+#include "store/sfs/types.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+namespace rgw::sal::sfs {
+
+SFSGC::~SFSGC() {
+  stop_processor();
+  finalize();
+}
+
+void SFSGC::initialize(CephContext *_cct, SFStore *_store) {
+  cct = _cct;
+  store = _store;
+}
+
+void SFSGC::finalize() {
+}
+
+int SFSGC::process() {
+  // This is the method that does the garbage collection.
+  auto max_objects = cct->_conf->rgw_gc_max_objs;
+  ldpp_dout(this, 20) << "garbage collection: processing with max_objects = "
+                    << max_objects << dendl;
+
+  // For now, delete only the objects with deleted bucket.
+  return process_deleted_buckets(max_objects);
+}
+
+bool SFSGC::going_down() {
+  return down_flag;
+}
+
+void SFSGC::start_processor() {
+  worker = new GCWorker(this, cct, this);
+  worker->create("rgw_gc");
+}
+
+void SFSGC::stop_processor() {
+  down_flag = true;
+  if (worker) {
+    worker->stop();
+    worker->join();
+    delete worker;
+  }
+  worker = nullptr;
+}
+
+unsigned SFSGC::get_subsys() const
+{
+  return dout_subsys;
+}
+
+std::ostream& SFSGC::gen_prefix(std::ostream& out) const
+{
+  return out << "garbage collection: ";
+}
+
+int SFSGC::process_deleted_buckets(long int &max_objects) {
```

Comment:
The value WAS in case we added more methods in the future deleting stuff.
What I did is move `max_objects` to be a member and I set the value from rgw config each iteration.
That way I don't need to pass max_objects around.